pkg/artifacts/signable.go

```diff
@@ -268,18 +272,34 @@ func extractTargetFromResults(obj objects.TektonObject, identifierSuffix string,
 		if strings.HasSuffix(res.Name, identifierSuffix) {
 			marker := strings.TrimSuffix(res.Name, identifierSuffix)
 			if v, ok := ss[marker]; ok {
+				if res.Value.StringVal == "" {
+					logger.Errorf("error getting string value for %s", res.Name)
+					continue
+				}
 				v.URI = strings.TrimSpace(res.Value.StringVal)
 
 			} else {
+				if res.Value.StringVal == "" {
+					logger.Errorf("error getting string value for %s", res.Name)
+					continue
+				}
 				ss[marker] = &StructuredSignable{URI: strings.TrimSpace(res.Value.StringVal)}
 			}
 			// TODO: add logic for Intoto signable target as input.
 		}
 		if strings.HasSuffix(res.Name, digestSuffix) {
 			marker := strings.TrimSuffix(res.Name, digestSuffix)
 			if v, ok := ss[marker]; ok {
+				if res.Value.StringVal == "" {
+					logger.Errorf("error getting string value for %s", res.Name)
+					continue
+				}
 				v.Digest = strings.TrimSpace(res.Value.StringVal)
 			} else {
+				if res.Value.StringVal == "" {
+					logger.Errorf("error getting string value for %s", res.Name)
+					continue
+				}
 				ss[marker] = &StructuredSignable{Digest: strings.TrimSpace(res.Value.StringVal)}
 			}
 		}
```

Comment:
nit: Seems like we only accept string type result here right? If so, is it possible that we move the `if res.Value.StringVal == "" {}` block right after the line `for _, res := range obj.GetResults() {` so that we don't repeat the check?

Reply:
That's a good point! My only concern is that we probably don't want to log this when the results don't have the right suffices. I can check the StringVal at where you mention, but don't log at all and keep continuing, but I definitely feel users would find the logged error being helpful

Reply:
That sgtm! If you want to keep the log in, perhaps downgrade to log.Debug? This definitely isn't log.Error worthy since it isn't really an error and it's not particularly actionable. log.Debug will remove it from production logs, but will still keep it around for local debugging, etc.

Reply:
Got it. +1 to log this error.

In that case, maybe we can just check the result type only after checking suffix each time instead of checking result type in both inner `if v, ok := ss[marker]; ok {` and its else block?

Reply:
That's a good point! Change is done!